lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/MultiRuntimeYarnBootstrapDriverConfigGenerator.java

```diff
@@ -0,0 +1,279 @@
+            .bindNamedParameter(TcpPortRangeCount.class, Integer.toString(appSubmissionParams.getTcpRangeCount()))
+            .bindNamedParameter(TcpPortRangeTryCount.class, Integer.toString(appSubmissionParams.getTcpTryCount()))
+            .bindNamedParameter(JobSubmissionDirectoryPrefix.class,
+                    yarnJobSubmissionParams.getJobSubmissionDirectoryPrefix().toString())
+            .bindImplementation(RuntimeClasspathProvider.class, YarnClasspathProvider.class)
+            .bindConstructor(YarnConfiguration.class, YarnConfigurationConstructor.class)
+            .build();
+
+    final Configuration driverConfiguration = Configurations.merge(
+            Constants.DRIVER_CONFIGURATION_WITH_HTTP_AND_NAMESERVER,
+            multiRuntimeDriverConfiguration.build(),
+            providerConfig);
+
+    // add restart configuration if needed
+    if (multiruntimeAppSubmissionParams.getYarnRuntimeAppParameters() != null &&
+            multiruntimeAppSubmissionParams.getYarnRuntimeAppParameters().getDriverRecoveryTimeout() > 0) {
+      LOG.log(Level.FINE, "Driver restart is enabled.");
+
+      final Configuration yarnDriverRestartConfiguration =
+              YarnDriverRestartConfiguration.CONF.build();
+
+      final Configuration driverRestartConfiguration =
+              DriverRestartConfiguration.CONF
+                      .set(DriverRestartConfiguration.ON_DRIVER_RESTARTED, JobDriver.RestartHandler.class)
+                      .set(DriverRestartConfiguration.ON_DRIVER_RESTART_CONTEXT_ACTIVE,
+                              JobDriver.DriverRestartActiveContextHandler.class)
+                      .set(DriverRestartConfiguration.ON_DRIVER_RESTART_TASK_RUNNING,
+                              JobDriver.DriverRestartRunningTaskHandler.class)
+                      .set(DriverRestartConfiguration.DRIVER_RESTART_EVALUATOR_RECOVERY_SECONDS,
+                              multiruntimeAppSubmissionParams.getYarnRuntimeAppParameters().getDriverRecoveryTimeout())
+                      .set(DriverRestartConfiguration.ON_DRIVER_RESTART_COMPLETED,
+                              JobDriver.DriverRestartCompletedHandler.class)
+                      .set(DriverRestartConfiguration.ON_DRIVER_RESTART_EVALUATOR_FAILED,
+                              JobDriver.DriverRestartFailedEvaluatorHandler.class)
+                      .build();
+
+      return Configurations.merge(driverConfiguration, yarnDriverRestartConfiguration, driverRestartConfiguration);
+    }
+
+    return driverConfiguration;
+  }
+
+
+  /**
+   * Writes the driver configuration files to the provided location.
+   * @param bootstrapJobArgsLocation The path for the job args file
+   * @param bootstrapAppArgsLocation The path for the app args file
+   * @throws IOException
+   * @return A path to the written driver configuration
+   */
+  String writeDriverConfigurationFile(final String bootstrapJobArgsLocation,
```

Comment:
No. This is what this class MultiRuntimeYarnBootstrapDriverConfigGenerator does. It uses aother serializers to generate serialize and write the configuration. I don't see a good reason to create a serializer that is just a composition of other serializers.